### DISCUSSION TITLE
Backing up / File Storage

### DISCUSSION THREAD

**Discussion by TheMartianObserver** [vincode-io/Zavala#77] [General]
I'll admit that I'm a bit naive when it comes to how Mac apps store data. In evaluating Zavala, I was looking at whether it would serve as a good data store for the outlines (like a note app), as opposed to simply using it as an editor. However, the current implementation requires that all files are in Zavala's "library", and it doesn't look like the files themselves are stored as plain text anywhere (that I could find at least!).

The reason I'm wondering about this are backups, and how one might be able to make sure that the data is app "independent" and "future proof". I use NotePlan and one benefit is that all notes entered into the system are stored as simple plain text on disk, easy to find and back up. Also future proof in that if NotePlan goes away, all my notes are easily accessible in plain text form.

Even if I were to use this as a simple outline editor, constantly having to import and export files is a bit bothersome. Any chance we'd be able to get a mode where the app just reads and writes files (OPML or even MD) stored on disk?

Thanks!

**Comment by vincode-io** [comment]
Using an internal database or library is necessary to implement some features that Zavala has. The biggest one is iCloud syncing. How would I keep things in sync if a user is editing one of Zavala's OPML files while Zavala isn't running?

I understand your concern though. I've lost data before because the application I was using was obsolete when I went back and tried to open the file. 

I think the best I can do is build a function that that exports the whole database as OPML files inside a zip file. You would have to run this yourself periodically until I add automation to Zavala. Then you could schedule archives with something external to Zavala.

**Reply by TheMartianObserver** [reply]
Thanks for the reply. I think that a function that can export the entire database at once would be fantastic! (Still, would be great to have a "edit-only" mode that doesn't require using the database, but I understand!).

**Reply by vincode-io** [reply]
It is pretty much a sure thing that I'll add the bulk export function. I want it to be easy for users to get their data out of the application if I get hit by a bus or something.

I'll think some more about editing OPML files directly in the filesystem. Today I was looking for a good markdown editor that just let me edit the files directly without having to import them into a library. Ulysses had an interesting function where it let you add a directory to the sidebar and edit the markdown files there with reduced functionality. Maybe something like that would work for Zavala.

**Reply by anqi970301** [reply]
I'd like to see if there's already some built-in features for bulk backup. I currently use zavala for text-only so it's safe to select all -> export to opml. Still it would be nicer with internal support.

**Reply by vincode-io** [reply]
@anqi970301 take a look at the sample Shortcuts in Help. There are Archive Account and Restore Account shortcuts that do what you are looking for.

**Reply by anqi970301** [reply]
@vincode-io Thanks! That's what I need.
